### ISSUE TITLE
Fix false positives for rule-set interpolation in selector-max-empty-lines

### ISSUE THREAD

**Issue by manovotny** [stylelint/stylelint#4684]
> Clearly describe the bug

When using an expression in a CSS-in-JS [template string literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals), stylelint will flag a false positive for `selector-max-empty-lines`.

I have also create a [demo](https://stylelint.io/demo#N4Igxg9gJgpiBcICWBbADhATgFwATDExgENsYBxAGwgCNjKBlbAT0pgF9cAzTCFXAOQBnFmygBaSOggA7GDOxCBAbgA6M1Bhy4iQmHh59BI1jAm79K9esgyRuKrXpNTuALy5CJMo7qNRMAAG6rihuAAkwBbY7NYyYbgAFtgolPghCZ4Q1JjwuDSUxGAA1mrxYbEyGaHE6eWZWTl5AO6JSGRlCZXVuGh1DaGQTTpmnRXqgWXqMAAeWniwXMQArpR4vs4ByiAANCBCzArEMwjgQkLiSDLiAFZCu+CyXEgA5qfAGaogmKswQl95D71XBfPRsMDYLDiFDHcQwdAscSUK5-AG4AAMGUq7BA7CAA) of the issue.

Please note, as soon as I add a single space, the error goes away. Very odd.

![Screen Shot 2020-04-02 at 5 22 43 PM](https://user-images.githubusercontent.com/446260/78305572-dc083380-7506-11ea-8dc4-d866c0e3e748.jpg)

> Which rule, if any, is the bug related to?

`selector-max-empty-lines`

> What code is needed to reproduce the bug?

```js
import {createGlobalStyle} from 'styled-components';
import reset from 'styled-reset';

const GlobalStyle = createGlobalStyle`
    ${reset}
 
    html {
      color: black;
    }

    a {
        color: white;
    }

    p {
        color: red;
    }
`;

export default GlobalStyle;
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-max-empty-lines": 0
  }
}
```

> Which version of stylelint are you using?

13.2.1

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint "**/*.js" --report-needless-disables"`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

CSS-in-JS

> What did you expect to happen?

No errors / warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged:

```shell
5:5 error Expected no more than 0 empty lines (selector-max-empty-lines)
```

**Comment by jeddy3** [comment]
@manovotny Thanks for the clearly written report, demo and for using the template.

I suspect the CSS-in-JS parser that stylelint uses [to support CSS-in-JS](https://stylelint.io/about/syntaxes) is incorrectly parsing the `$reset` interpolation as part of the `html` selector. I've added this issue to https://github.com/stylelint/stylelint/issues/4574, which is a growing list of the problems with the syntax that we need help to fix.

The good news is that we intend to start using [our fork](https://github.com/stylelint/postcss-css-in-js) of the parser in the [next stylelint release](https://github.com/stylelint/stylelint/issues/4687). Fixes can now be made to our fork.


Please consider [contributing to the parser](https://stylelint.io/developer-guide/syntaxes) if you have time.

**Comment by manovotny** [comment]
FWIW, I no longer see this issue in `v13.4.0`.

**Comment by jeddy3** [comment]
@manovotny Thanks for letting us know.

I can't reproduce it [using the demo either](https://stylelint.io/demo#N4Igxg9gJgpiBcICWBbADhATgFwATDExgENsYBxAGwgCNjKBlbAT0pgF9cAzTCFXAOQBnFmygBaSOggA7GDOxCBAbgA6M1Bhy4iQmHh59BI1jAm79K9esgyRuKrXpNTuALy5CJMo7qNRMAAG6rihuAAkwBbY7CEhYQAW2CiU+PFhnhDUmPC4NJTEYADWajIZsTLpuMRpZRlhkNm5AO4JSGSl5dZ1oWi19Q1ZWLlEUJ1hFYGl6jAAHlp4sFzEAK6UeL7OAcogADQgQswKxLMI4EJC4kgy4gBWQnvgslxIAOZnwPGqIJhrMELfXKfHrfPRsMDYLDiFAncQwdAscSUa7-QG4AAM8Qq7BA7CAA) now.

I assume it was indirectly fixed by https://github.com/stylelint/postcss-css-in-js/pull/17, which was released in [`13.3.1`](https://stylelint.io/changelog#1331).

I'll close the issue.
